codemate/block.py

```diff
@@ -330,26 +331,34 @@ def use_black(self) -> str:
 
         Returns:
             str: The block syntax.
+
+        Raises:
+            InputError: When the generated Python 3 code isn't valid.
         """
+        self.validate()
         return black.format_str(self.syntax(), mode=black.Mode())
 
-    def validate(self) -> bool:
+    def validate(self, raise_error: bool = True) -> bool:
         """
-        Checks if the generated syntax structure is valid.
-        If not raises 'InputError' error that wraps black.InvalidInput error.
+        Checks if the generated syntax is Python 3 valid.
+
+        Args:
+            raise_error (bool): When True and the syntax is invalid, an exception will be
+                raised. When False and the syntax is invalid, False will be returned.
 
         Returns:
-            bool: True the generated syntax is valid.
+            bool: True when the syntax is valid, otherwise False.
 
         Raises:
-            InputError: When the generated Python code isn't valid by black.
+            InputError: When the generated Python 3 code isn't valid.
         """
         try:
-            self.use_black()
-        except black.InvalidInput as error:
-            raise InputError(error) from None
-        else:
-            return True
+            validator.validate(self.syntax(), raise_error=True)
```

Comment:
Don't you want to pass `raise_error` from the input?

Reply:
Correct my mistake

Reply:
We want always to raise the error.
If the syntax is invalid and the user doesn't want that we will raise the error, we catch it and returns False.